content/automate/ManageIQ/System/Event/StateMachines/Refresh.class/__methods__/check_refreshed.rb

```diff
@@ -0,0 +1,62 @@
+#
+# Description: This method checks the refresh completion.
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Event
+        module StateMachines
+          module Refresh
+            class CheckRefreshed
+              STATE_FINISHED = 'Finished'.freeze
+
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                check_status(refresh_task)
+              end
+
+              private
+
+              def check_status(task)
+                case task.state
+                when STATE_FINISHED
+                  @handle.root['ae_result'] = 'ok'
```

Comment:
@lfu If the State == "Finished" but Status != "Ok" we should set the ae_result as error and fail the state machine. Otherwise if there is a password failure during refresh we will continue processing the next state which is going to run the policy action.